transport/src/main/java/io/netty/channel/AbstractCoalescingBufferQueue.java

```diff
@@ -196,6 +197,19 @@ public final ByteBuf remove(ByteBufAllocator alloc, int bytes, ChannelPromise ag
                 }
             }
         } catch (Throwable cause) {
+            // Always decrement to keep things consistent. We decrement directly here and not in a finally-block
+            // to ensure that the state is consistent even if it would be accessed via a listener that is
+            // attached to the promise that we fail below.
+            decrementReadableBytes(originalBytes - bytes);
+
+            if (entry instanceof ByteBuf) {
```

Comment:
Would this be better by using `entryBuffer`?

Reply:
Honestly I think we can jus remove the whole instanceof dance and just peek directly